docs/configuration.md

```diff
@@ -0,0 +1,261 @@
+- It needs to precisely control when connectors start and stop. 
+- It needs to reset the entire connector configuration without restarting 
+  the application.
+
+### Registering and Using a Named Connector
+
+The application calls `ConnectorRegistry.register()` to register the named
+connector configuration.
+
+```java
+GoogleCredentials myCredentials = GoogleCredentials.create(authToken);
+
+ConnectorConfig config = new ConnectorConfig.Builder()
+  .withTargetPrincipal("example@project.iam.googleapis.com")
+  .withDelegates(Arrays.asList("delegate@project.iam.googleapis.com"))
+  .withGoogleCredentials(myCredentials)
+  .build();
+
+ConnectorRegistry.register("my-connector",config);
+```
+
+Then the application tells a database connection to use a named connector by 
+adding the `cloudSqlNamedConnector` to the JDBC connection properties. 
+
+It can do this by adding `cloudSqlNamedConnector` to the JDBC URL:
+
+```java
+String jdbcUrl = "jdbc:mysql:///<DATABASE_NAME>?"+
+    +"cloudSqlInstance=project:region:instance"
+    +"&cloudSqlNamedConnector=my-connector"
+    +"&socketFactory=com.google.cloud.sql.mysql.SocketFactory"
+    +"&user=<DB_USER>&password=<PASSWORD>";
+```
+
+Or by adding `cloudSqlNamedConnector` to the JDBC connection properties:
+
+```java
+// Set up URL parameters
+Properties connProps = new Properties();
+
+connProps.setProperty("user","<DB_USER>");
+connProps.setProperty("password","<PASSWORD>");
+connProps.setProperty("sslmode","disable");
+connProps.setProperty("socketFactory","<DRIVER_CLASS>");
+connProps.setProperty("cloudSqlInstance","project:region:instance");
+
+connProps.setProperty("cloudSqlNamedConnector","my-connector");
+
+// Initialize connection pool
+HikariConfig config = new HikariConfig();
+config.setJdbcUrl("jdbc:mysql:///<DB_NAME>");
+config.setDataSourceProperties(connProps);
+config.setConnectionTimeout(10000); // 10s
+
+HikariDataSource connectionPool = new HikariDataSource(config);
+```
+
+In R2DBC, use the `NAMED_CONNECTOR` ConnectionFactory option.
+
+When using a named connector, the JDBC connection will ignore all connector
+configuration properties in the JDBC connection properties, using the named
+connector configuration instead. See the full list
+of [connector configuration properties](#connector-configuration) below.
+
+The
+test [JdbcPostgresNamedConnectorIntegrationTest.java](../jdbc/postgres/src/test/java/com/google/cloud/sql/postgres/JdbcPostgresNamedConnectorIntegrationTests.java)
+contains a full example showing how to use a named connector.
+
+### Closing Named Connectors
+
+
+You can close a named connector by calling `ConnectorRegistry.close()`. This
+will stop the certificate refresh process for that connector. It will also
+cause all subsequent attempts connect using this named connector to fail.
+
+```java
+ConnectorRegistry.close("my-connector");
+```
+
+Existing open database `Connection` instances will continue work until they
```

Comment:
Nit: we're not naming a class when we say "Connection" so let's just make this:

"Existing open database connections will continue to work ..."